runtime/shared_common/CacheLifecycleManager.cpp

```diff
@@ -317,19 +317,19 @@ printSharedCache(void* element, void* param)
 			j9tty_printf(PORTLIB, "%-7s", "");
 		}
 #if !defined(WIN32)
-		if (currentItem->os_shmid != (UDATA)J9SH_OSCACHE_UNKNOWN) {
+		if (currentItem->os_shmid != J9SH_OSCACHE_UNKNOWN) {
 			j9tty_printf(PORTLIB, "%-15d", currentItem->os_shmid);
 		} else {
 			j9tty_printf(PORTLIB, "%-15s", "");
 		}
-		if (currentItem->os_semid != (UDATA)J9SH_OSCACHE_UNKNOWN) {
+		if (currentItem->os_semid != J9SH_OSCACHE_UNKNOWN) {
 			j9tty_printf(PORTLIB, "%-15d", currentItem->os_semid);
 		} else {
 			j9tty_printf(PORTLIB, "%-15s", "");
 		}
 #endif
 		if (currentItem->nattach == 0) {
-			if (J9SH_OSCACHE_UNKNOWN == currentItem->lastdetach) {
+			if ((I_64)J9SH_OSCACHE_UNKNOWN == currentItem->lastdetach) {
```

Comment:
The condition should be `J9SH_OSCACHE_UNKNOWN == (U_64)currentItem->lastdetach`.

Reply:
Is this going to work when UDATA is 32-bits? I left it as `(IDATA)J9SH_OSCACHE_UNKNOWN` for now.

Reply:
I expect a picky 32-bit compiler (i.e. msvc) to complain about the direct cast of `UDATA` to `I_64` which changes both the signed-ness and the width, thus my suggestion to put the cast on the other side (casting `UDATA` to `U_64`).